asyn/asynPortDriver/paramVal.h

```diff
@@ -54,7 +56,7 @@ class paramVal {
         epicsInt32   ival;
         epicsUInt32  uival;
         epicsFloat64 dval;
-        char         *sval;
+        std::string  *sval;
```

Comment:
yuck.  I'd suggest just moving ```sval``` outside the union as a non-pointer member.  This is a bit bigger, but not much.  I think this is offset by the code simplification/safety and avoiding an extra indirection.  For me (linux on x86_64 w/ gcc 4.9.2) ```sizeof(std::string)==8```.